src/gluonts/model/deepar/_network.py

```diff
@@ -425,6 +460,17 @@ def hybrid_forward(
         # need to mask possible nans and -inf
         loss = F.where(condition=loss_weights, x=loss, y=F.zeros_like(loss))
 
+        # rnn_outputs is already merged into a single tensor
+        assert not isinstance(rnn_outputs, list)
+        # it seems that the trainer only uses the first return value for backward
+        # so we only add regularization to weighted_loss
+        if self.alpha:
+            ar_loss = self.ar_loss(rnn_outputs)
+            weighted_loss = F.elemwise_add(weighted_loss, ar_loss)
```

Comment:
Just curious: why not `weighted_loss + ar_loss`? (Similarly elsewhere)

Reply:
I think they are equivalent